### ISSUE TITLE
Allow magic numbers in simple method returns

### ISSUE THREAD

**Issue by h-lame** [Bodacious/no-magic-numbers-cop#54]
I'm not high conviction on this, I'm not sure I'm into the no returns cop, at least in the simple case.  E.g. I think that the following code:

```ruby
def my_magic_number
  6
end
```

or

```ruby
def my_magic_number = 6
```

shouldn't be an offence.  We've given that magic number a name - but a method rather than a constant.  Not sure we'd gain more from it being:

```ruby
MY_MAGIC_NUMBER = 6

def my_magic_number
  MY_MAGIC_NUMBER
end
```

or

```ruby
MY_MAGIC_NUMBER = 6

def my_magic_number = MY_MAGIC_NUMBER
```

I reckon a method is more flexible than a constant in terms of allowing for mutation later.  I'd rather code used `some_obj.my_magic_number` vs. `SomeClass::MY_MAGIC_NUMBER` because it's easier to change `my_magic_number`s implementation to a calculation if we need to (YAGNI Violation! - 1,000 years dungeon for you Murray!).

**Comment by Bodacious** [comment]
@h-lame Thank you for opening this issue.

To be honest, this is something I've dithered back and forth about. 

I can think of cases where a method is beneficial in place of a constant, but I still tend to lean towards the method returning a constant rather than a number—assuming the method name might be different from the constant name.

The case I came up with in my head was a bit contrived:

``` ruby 
class NullAccount 
  def balance
    0
  end
end

class User 
  def account 
    Account.find(account_id) || NullAccount.new 
  end
end

@user = User.new(account_id: nil)
@user.balance # => 0
``` 

... but I _think_ this is still better written with NullAccount#balance returning a constant that explains why it's zero in the first place? 

In any case, I've created a PR (#56) that will allow teams to configure this based on their own preference, as I can imagine the current default might be controversial.

WDYT?
